lib/kontena/plugin/vagrant/master/restart_command.rb

```diff
@@ -4,17 +4,20 @@ class RestartCommand < Kontena::Command
 
     def execute
       require_relative '../../../machine/vagrant'
-      
+
       vagrant_path = "#{Dir.home}/.kontena/vagrant_master"
       abort("Cannot find Vagrant kontena-master".colorize(:red)) unless Dir.exist?(vagrant_path)
       Dir.chdir(vagrant_path) do
-        spinner "Restarting Vagrant kontena-master " do
-          Open3.popen2('vagrant reload') do |stdin, output, wait|
-            while o = output.gets
-              print o if ENV['DEBUG']
-            end
+        spinner "Executing 'vagrant reload' for kontena-master"
+        Open3.popen2('vagrant reload') do |stdin, output, wait|
```

Comment:
Ah, these are in the plugin command class, not the provisioner classes... just call something like `system || return false` directly instead... returning `false` from the `Kontena::Command#execute` will allow the CLI command to exit with an error.